spec/unit/provider/exec/pwsh_spec.rb

```diff
@@ -39,80 +33,84 @@
       end
 
+      let(:shell_command) { Puppet.features.microsoft_windows? ? 'cmd.exe /c' : '/bin/sh -c' }
+
       it "should call exec run" do
         Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run)
 
         provider.run_spec_override(command)
       end
 
-      context "on windows", :if => Puppet.features.microsoft_windows? do
-        it "should call cmd.exe /c" do
-          Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run)
-            .with(regexp_matches(/^cmd.exe \/c/), anything)
+      it "should call shell command" do
+        Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run)
+          .with(regexp_matches(/#{shell_command}/), anything)
 
-          provider.run_spec_override(command)
-        end
+        provider.run_spec_override(command)
+      end
 
-        it "should quote the path to the temp file" do
-          path = 'C:\Users\albert\AppData\Local\Temp\puppet-powershell20130715-788-1n66f2j.ps1'
+      it "should quote the path to the temp file", :if => Puppet.features.microsoft_windows? do
+        # Path quoting is only required on Windows
+        path = 'C:\Users\albert\AppData\Local\Temp\puppet-powershell20130715-788-1n66f2j.ps1'
 
-          provider.expects(:write_script).with(command).yields(path)
-          Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run).
-            with(regexp_matches(/^cmd.exe \/c ".* < "#{Regexp.escape(path)}""/), false)
+        provider.expects(:write_script).with(command).yields(path)
+        Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run).
+          with(regexp_matches(/^#{shell_command} .* < "#{Regexp.escape(path)}"/), false)
 
-          provider.run_spec_override(command)
-        end
+        provider.run_spec_override(command)
+      end
 
-        it "should supply default arguments to supress user interaction" do
-          Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run).
-            with(regexp_matches(/^cmd.exe \/c ".* #{args} < .*"/), false)
+      it "should supply default arguments to supress user interaction" do
+        Puppet::Type::Exec::ProviderPwsh.any_instance.expects(:run).
+          with(regexp_matches(/^#{shell_command} .* #{args} < .*"/), false)
 
-          provider.run_spec_override(command)
-        end
+        provider.run_spec_override(command)
       end
     end
 
     context "actual runs" do
-      context "on Windows", :if => Puppet.features.microsoft_windows? do
-        it "returns the output and status" do
-          output, status = provider.run(command)
+      # The usage of uname is a little fragile however there is basically nothing
+      # which is universal across all Linux/Unix/Mac distributions; Unlike Well Known SIDS in Windows
+      # The closest is the presence of the uname command and its generic text output
+      let(:command) { Puppet.features.microsoft_windows? ?
+        '$(Get-CIMInstance Win32_Account -Filter "SID=\'S-1-5-18\'") | Format-List' :
```

Comment:
Not sure if it's worth doing... but can't we just use `Get-CimInstance` on all platforms (with a different type I suppose)?

Reply:
Not sure.  Will need to look into it more.

Reply:
Ah no... looks like CIM classes aren't part of PowerShell itself - so might not be anything there to choose from anyhow.